src/foam/lib/PropertyPredicate.js

```diff
@@ -0,0 +1,26 @@
+foam.INTERFACE({
+  package: 'foam.lib',
+  name: 'PropertyPredicate',
+
+  methods: [
+    {
+      name: 'propertyPredicateCheck',
+      type: 'Boolean',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        },
+        {
+          name: 'fo',
```

Comment:
What is this field used for?

Reply:
But we could roll the old vs new check into the predicate right?  Adding a predicate for ValueUpdated.  Then a delta JSON outputter could just be implemented as `AND(ValueChanged, Permissioned, StorageTransient)` rather than having a separate stringifyDelta method.

Would be a cleaner interface.  We should also put a better name for this like "obj" instead of "fo"